### PR TITLE
SOLR-15784 Remove SolrJ dep on commons-io

### PR DESCRIPTION
**Pull request by madrob**
https://issues.apache.org/jira/browse/SOLR-15784

### CODE REVIEW

File: solr/solrj/src/java/org/apache/solr/common/EmptyEntityResolver.java [hunk 1]
```diff
@@ -17,15 +17,15 @@
 package org.apache.solr.common;
 
 import java.io.InputStream;
+
+import org.apache.http.impl.io.EmptyInputStream;
```
Comment (dsmiley) [inline solr/solrj/src/java/org/apache/solr/common/EmptyEntityResolver.java:21]: Maybe okay for now but we may soon be looking to not depend needlessly on Apache HttpClient

**Review by dsmiley — APPROVED** [review]
Nice :-)

File: solr/test-framework/src/java/org/apache/solr/SolrJettyTestBase.java [hunk 2]
```diff
@@ -172,39 +179,46 @@ public static String legacyExampleCollection1SolrHome() {
...
-      org.apache.commons.io.FileUtils.forceMkdir(collection1Dir);
+      Path tempSolrHome = LuceneTestCase.createTempDir();
+      Files.copy(Path.of(sourceHome, "server", "solr", "solr.xml"), tempSolrHome.resolve("solr.xml"));
+
+      Path collection1Dir = tempSolrHome.resolve("collection1");
+      Files.createDirectories(collection1Dir.resolve("conf"));
+
+      Path configSetDir = Path.of(sourceHome, "server", "solr", "configsets", "sample_techproducts_configs", "conf");
+
+      Files.walkFileTree(configSetDir, new SimpleFileVisitor<>() {
```
Comment (janhoy) [inline solr/test-framework/src/java/org/apache/solr/SolrJettyTestBase.java:191]: Should we move this copy-files code to a Solr Utils class?

**Review by janhoy — COMMENTED** [review]
Great cleanup. Just a minor comment

File: solr/solrj/src/java/org/apache/solr/common/EmptyEntityResolver.java (hunk 1 above)
Comment (madrob) [reply solr/solrj/src/java/org/apache/solr/common/EmptyEntityResolver.java:21]: I’m happy to kick this can down the road

**Pull request merged**